### ISSUE TITLE
Analysing uncertainty in NMT

### ISSUE THREAD

**Issue by haorannlp** [pytorch/fairseq#3032]
### Before asking:   
1. search the issues.   
2. search the docs.    

#### What is your question?
Hi @myleott,
I'm very interested in your paper "Analyzing Uncertainty in Neural Machine Translation" and tried to replicate the  ``cumulative sequence probability of hypos obtained by beam search`` in Figure 1 left.
I generated hypos using this command:
``python generate.py data_path --path model_path --batch-size 512 --beam 5 --nbest 5 --remove-bpe``
The log2 positional scores of each of 5 hypothesis are summed up and then we compute the 2^(sum) to get 5 probabilities.
In my understanding, the sum of the 5 probabilities should be the cumulative seq probability of 5 hypos.
Unfortunately, this cumulative prob. is much smaller than in the paper, 0.01 vs 0.15
I used WMT4 EN-DE instead of EN-FR but I think this should not make much difference. 

Did I miss something here? Thank you.

#### What's your environment?

 - fairseq Version (e.g., 1.0 or master):0.9
 - PyTorch Version (e.g., 1.0): 1.2
 - OS (e.g., Linux): linux
 - How you installed fairseq (`pip`, source):
 - Build command you used (if compiling from source):
 - Python version:
 - CUDA/cuDNN version:
 - GPU models and configuration:
 - Any other relevant information:

**Comment by myleott** [comment]
Great question! I went back over my scripts/results from that time (thankful for backups 😄) and it turns out that 0.15 is correct for the WMT'14 En-Fr model and **validation** set that we used, which is 26K sentences held-out from the training set. However, you are also correct that for some other models it's closer to 0.01.

You can find the model checkpoint and validation set here: https://dl.fbaipublicfiles.com/fairseq/models/repro_analyzing_uncertainty.wmt14_en_fr.tar.gz

Since the model file is very old, it requires a small change to be able to load the model, which I've put in the [repro_analyzing_uncertainty](https://github.com/pytorch/fairseq/compare/repro_analyzing_uncertainty?expand=1) branch:
```bash
$ git checkout repro_analyzing_uncertainty
$ PYTHONPATH=. python fairseq_cli/generate.py \
    repro_analyzing_uncertainty.wmt14_en_fr/data-bin \
    --path repro_analyzing_uncertainty.wmt14_en_fr/model.pt \
    --beam 5 --nbest 5 --remove-bpe --gen-subset valid
(...)
avg cumulative prob tensor(0.1463, device='cuda:0')
2020-12-15 11:57:44 | INFO | fairseq_cli.generate | NOTE: hypothesis and token scores are output in base 2
2020-12-15 11:57:44 | INFO | fairseq_cli.generate | Translated 26658 sentences (966478 tokens) in 264.5s (100.77 sentences/s, 3653.39 tokens/s)
Generate valid with beam=5: BLEU4 = 47.71, 68.9/51.6/41.8/34.8 (BP=1.000, ratio=1.001, syslen=869495, reflen=868288)
```

**Comment by haorannlp** [comment]
@myleott Thanks.  In WMT'14 EN-DE, if we use newstest2013 as valid set, the sequence level probability is ~0.02(beam 5), ~0.04(beam 100)
